src/sentry/processing/backpressure/memory.py

```diff
@@ -57,8 +55,12 @@ def iter_cluster_memory_usage(cluster: Cluster) -> Generator[ServiceMemory, None
     """
     if isinstance(cluster, RedisCluster):
         # `RedisCluster` returns these as a dictionary, with the node-id as key
-        for info in cluster.info().values():
-            yield get_memory_usage(info)
+        cluster_info = cluster.info()
     else:
-        # otherwise, lets just hope that `info()` does the right thing
-        yield get_memory_usage(cluster.info())
+        # rb.Cluster returns a promise with a dictionary with a _local_ node-id as key
+        with cluster.all() as client:
+            promise = client.info()
+        cluster_info = promise.value
```

Comment:
I don’t fully understand what this does/returns, how is that turned into something that iterates over all the nodes?

Reply:
See the `rb` docs ~ https://rb.readthedocs.io/en/latest/#rb.Cluster.all ~ it does fan out to all nodes and returns a promise object with the result in the `value` attribute. Another example is in this section, but there it iterates over the promise instead of accessing `.value` ~ https://rb.readthedocs.io/en/latest/#routing